Jenkinsfile

```diff
@@ -1,69 +1,98 @@
+        rename_license = "${postgres_dir}/license.txt"
+    }
+
+parameters {
+    string(defaultValue: "master", description: 'What version?', name: 'version_name')
+    choice(choices: ['branch', 'tag'], description: "Branch or Tag?", name: 'version_type')
+}
+
+stages {
+    stage('Build Info') {
+        steps {
+            echo "Version: ${params.version_name}\nVersion Type: ${params.version_type}\nCommit: ${env.GIT_COMMIT}"
+        }
+    }
     stage('Install') {
-        sh "sudo dnf -y install origin-clients"
-        sh "rpm -q docker"
-        sh "which docker"
-        sh "echo OPTIONS=\\'--log-driver=journald\\' > /tmp/docker.conf"
-        sh "echo DOCKER_CERT_PATH=/etc/docker >> /tmp/docker.conf"
-        sh "sudo cp /tmp/docker.conf /etc/sysconfig/docker"
-        sh "cat /etc/sysconfig/docker"
-        sh "sudo systemctl start docker"
-        checkout scm
-        sh "sleep 35s"
-        sh "ps aux | grep docker"
-        sh "sudo docker -v"
-        sh "sudo setenforce 0"
+        steps {
+            sh "sudo dnf -y install origin-clients"
+            sh "rpm -q docker"
+            sh "which docker"
+            sh "echo OPTIONS=\\'--log-driver=journald\\' > /tmp/docker.conf"
+            sh "echo DOCKER_CERT_PATH=/etc/docker >> /tmp/docker.conf"
+            sh "sudo cp /tmp/docker.conf /etc/sysconfig/docker"
+            sh "cat /etc/sysconfig/docker"
+            sh "sudo systemctl start docker"
+
+            // checkout scm
+            script {
+                if ("${params.version_type}" == 'branch') {
+                    echo "Checkout Branch"
+                    checkout changelog: false, poll: false, scm: [$class: 'GitSCM', branches: [[name: "${params.version_name}"]], doGenerateSubmoduleConfigurations: false, extensions: [], submoduleCfg: [], userRemoteConfigs: [[url: "${env.GIT_URL}"]]]
+                }
+                if ("${params.version_type}" == 'tag') {
+                    echo "Checkout Tag"
+                    checkout changelog: false, poll: false, scm: [$class: 'GitSCM', branches: [[name: "refs/tags/${params.version_name}"]], doGenerateSubmoduleConfigurations: false, extensions: [], submoduleCfg: [], userRemoteConfigs: [[url: "${env.GIT_URL}"]]]
+                }
+              }
+
+            sh "sleep 35s"
+            sh "ps aux | grep docker"
+            sh "sudo docker -v"
+            sh "sudo setenforce 0"
+        }
     }
     stage('Copy UI into Server') {
-        copyArtifacts filter: 'quipucords-ui-dist.tar.gz', fingerprintArtifacts: true, projectName: "qpc_${VERSION}_ui_distribution", selector: lastCompleted()
-        sh "tar -xvf quipucords-ui-dist.tar.gz"
-        sh "cp -rf dist/client quipucords/"
-    	sh "cp -rf dist/templates quipucords/quipucords/"
-        sh "rm -rf dist"
+        steps {
+            copyArtifacts filter: 'quipucords-ui.*.tar.gz', fingerprintArtifacts: true, projectName: "quipucords-ui-build-job", selector: lastCompleted()
```

Comment:
Does this need to be `quipucords-ui-dist.tar.gz`?  The other UI jenkins job uses `quipucords-ui-dist.tar.gz`.  This depends on whether you updated the UI jenkins artifact to match that pattern.

Reply:
The UI job will be `quipucords-ui.*.tar.gz` once that PR ([UI #59](https://github.com/quipucords/quipucords-ui/pull/59/)) is merged. I want to merge it with since one so they don't break each other.